frontend/src/metabase/lib/entities.js

```diff
@@ -392,7 +392,11 @@ export function createEntity(def) {
     [getEntities, getEntityId],
     (entities, entityId) => denormalize(entityId, entity.schema, entities),
   )((state, { entityId } = {}) =>
-    typeof entityId === "object" ? JSON.stringify(entityId) : entityId,
+    typeof entityId === "object"
+      ? JSON.stringify(entityId)
+      : entityId
+      ? entityId
+      : "",
```

Comment:
This change fixes the warning. the function never returns "undefined".

Note this solution of returning an empty string isn't new - we do the same thing on line 436 below.